### ISSUE TITLE
Feature Request: Title Tooltip Field

### ISSUE THREAD

**Issue by kylebrandt** [grafana/grafana#4079]
I would find it useful to have a field for text that would display as a tooltip when hovering over the title of graph panel.

My use case is that in bosun each metric has a description field. Those descriptions give more information about the metric. These can be long, so there doesn't seem to be a current space on grafana dashboards for this. The best I have come up with for a workaround is to create a drill down link that has no link value (Just a lot of text for the title). Which is a little silly and ends up looking like this:

![image](https://cloud.githubusercontent.com/assets/1692624/13177949/51ca0464-d6ea-11e5-96f1-7f1dab50f495.png)

Maybe there is a better way that I am missing?

**Comment by utkarshcmu** [comment]
Check this out for screenshots : https://github.com/grafana/grafana/issues/1505
The work was done here in this PR : https://github.com/grafana/grafana/pull/3691
But it was not merged into Grafana master in favor of https://github.com/grafana/grafana/issues/1168

Still if you want tooltip with Title as shown in #1505 , its available here [https://github.com/justforkit/grafana]

**Comment by utkarshcmu** [comment]
Anyway, thanks for the great work in Bosun. It is very useful. :+1:

**Comment by kylebrandt** [comment]
Hmm... So the one that was merged isn't collapsed by default?

**Comment by utkarshcmu** [comment]
Sorry, I did not understand your question. #3691 was never merged and was rejected in Grafana master. #3691 is merged in my fork [https://github.com/justforkit/grafana].

**Comment by kylebrandt** [comment]
I meant the notes in #1168 - seems like the notes are not collapsed based on your comment, so with lots of graphs or long notes it gets messy?

**Comment by utkarshcmu** [comment]
Yes it will get somewhat messy, but according to @torkelo in #1505 , the panel notes/info should be visible all the time. Anyway, I am still completely in favor of something hidden and which only shows up on the dashboard when needed (either by hovered tooltip or collapsed notes). Dashboard space is precious for my use case at least. :)

**Comment by kylebrandt** [comment]
@torkelo Request for reconsideration on

> Having panel description text that only appear on mouse over (like tooltip) is not something that feels like a good idea. If the information is important then it should be visibile all the time.

The reason is because there are two different user stories here:

1. Focused dashboards with important curated graphs. In this scenario, I think you are correct. You don't want to hide important information that is essential to interpreting the visualization. 
2. Many graph dashboards: In this scenario, there are many graphs (say, row X columns = 6x6). In this case, I'm basically visually looking for things that seem anomalous. Once I see those things, I then want more information about the metric / panel. Always having the notes displayed reduces the number of panels I can have on my screen for this sort of thing overview and makes it harder to read.

**Comment by utkarshcmu** [comment]
@torkelo 

Would you like to reconsider #1505 & #3691 as you labeled this issue as "help wanted"?

**Comment by torkelo** [comment]
Yes, but need to use new drop component instead of the normal tooltip

**Comment by utkarshcmu** [comment]
@torkelo - Can you please give me an example from the code where this new drop component is used? Also, if you can briefly describe how you want the dropdown of this help text. Just wanted to make sure whats your vision before starting any work on this. :)

**Comment by torkelo** [comment]
the new tooltip / popover component is used by annotations and the help popover (used in data source edit view, when you focus datasource name or http url inputs )

**Comment by utkarshcmu** [comment]
This help text is optional. It will show up "i" icon only when there is any help text otherwise not.

<img width="1423" alt="screen shot 2016-03-01 at 11 09 28 pm" src="https://cloud.githubusercontent.com/assets/1019609/13453190/e3c2ecf2-e002-11e5-86c0-441df40875e4.png">

<img width="1430" alt="screen shot 2016-03-01 at 11 09 13 pm" src="https://cloud.githubusercontent.com/assets/1019609/13453202/faccefba-e002-11e5-9dcb-d84236a6dc2b.png">

Any feedback?

**Comment by torkelo** [comment]
@utkarshcmu looks good, how does it handle out of bounds scenarios (long text on panel close to right edge of viewport). Currently css for the help popover is that is gets hidden when out of bounds, the drop component also supports moving attachment position when out of bounds, but due to the current css for the help popover this is disabled as it is hidden.

**Comment by torkelo** [comment]
@mattttt thoughts ^

**Comment by utkarshcmu** [comment]
This is a screenshot with a help text of 208 characters. 
<img width="766" alt="screen shot 2016-03-02 at 12 56 35 am" src="https://cloud.githubusercontent.com/assets/1019609/13455363/d40a5886-e011-11e5-81e3-050dd1894530.png">

Character count = 332
<img width="853" alt="screen shot 2016-03-02 at 1 00 40 am" src="https://cloud.githubusercontent.com/assets/1019609/13455425/3c8656c6-e012-11e5-83a2-9597e9c96a87.png">

IMO this serves the basic purpose of help text. If the help text is too long, i.e. more than 200-300 characters then that information is too verbose.

**Comment by mattttt** [comment]
Not near wifi at the moment, will look later tonight or worst case tomorrow.

**Comment by utkarshcmu** [comment]
@mattttt Any updates on this?

**Comment by sbengo** [comment]
+1! Would be awesome to add info about shown metrics for the final user!

**Comment by toni-moreno** [comment]
:+1: 

We need this PR to show a good description on the metrics shown to users.

 We have had lots of misunderstanding issues with users that can not understand what they are watching in some panels.

**Comment by dutchiechris** [comment]
+1 I have a set of default dashboards as part of NetApp storage monitoring integration and users often don't understand exactly what they are looking at, and since I am not their colleague I can't easily answer!  A help field would help the user greatly and as the native/default dashboard feature of Grafana becomes more used it is helpful there too.

**Comment by utkarshcmu** [comment]
If this can become part of 3.0.0 , it would be awesome. This will be so helpful for the viewers to understand the graphs. :)

**Comment by merwy** [comment]
+1 I would really need panel tooltips on my dashboards as I have some specific and tricky panels which need some additional explanation about what exactly they present. Looking forward for this feature!

**Comment by utkarshcmu** [comment]
@mattttt Any update on this issue / PR #4224

**Comment by utkarshcmu** [comment]
@torkelo / @bergquist I think this functionality is needed by many. Can you guys take a look at #4224 ?

**Comment by mattttt** [comment]
I very much like the revised tool tip, but because space conflicts with the drill down, it'll ned a bit more thought.  

That said, because this entire row header needs a re-think, I'm inclined to approve this design right now (with the expectation that few panels will have both info and drilldown) while we focus on refactoring to support new features/functionality. 

👍

**Comment by utkarshcmu** [comment]
Thanks @mattttt , will redo the PR #4224 asap!

**Comment by utkarshcmu** [comment]
Screenshot:

<img width="876" alt="screen shot 2016-12-06 at 3 54 02 am" src="https://cloud.githubusercontent.com/assets/1019609/20924957/ddd35efc-bb68-11e6-9635-137ecb7b7b85.png">

**Comment by torkelo** [comment]
Made progress on refactoring PR #6847 

Tried to improve layout for panel general tab to fit a text area for panel description (help text). Want text area as I would like to support markdown there instead of sanitizing html. 

The thing that needs some thought is how to present this icon, right now we have 3 different icons next to panel title. Is there a way to organize them better (all before or after title for example, seperator?). @mattttt any thoughts? 

![image](https://cloud.githubusercontent.com/assets/10999/21190549/73e1f664-c222-11e6-8867-f782efa4eb49.png)

**Comment by utkarshcmu** [comment]
@torkelo/@matt One way to not have the "circle-i" icon is to display description when a user hovers over the Panel title but I am not sure if that is ideal or if we need an extra icon to show description.

**Comment by torkelo** [comment]
I think the icon is important to signal that there is a panel description there

**Comment by torkelo** [comment]
Made some good progress on this, trying to unify panel drilldown links and panel description text into the same area, the same are used for error info as well. 

![panel_desc](https://cloud.githubusercontent.com/assets/10999/21263105/272ba9e0-c395-11e6-9118-146f57c06cf2.gif)

For full video where I talk through it:
http://grafana.org/assets/videos/panel_description.mp4

**Comment by utkarshcmu** [comment]
@torkelo This looks great! 👍🏽

**Comment by dutchiechris** [comment]
I loaded grafana:master from docker (commit: 0a0b4f5 shown on login page) to check out this feature and added a data source and panel with valid metric, and then added a description on the edit:general tab and I don't get any info icon in the corner.  I saved, reloaded, tried in Firefox and Chrome, and verified it's in the json response, but didn't see it yet.  Any ideas?

**Comment by utkarshcmu** [comment]
In the current master I see the description but the corner is without the blue background which is harder to spot. So, either we make the background blue of the corner or make the "i" whiter or bolder.

<img width="273" alt="screen shot 2016-12-18 at 3 03 45 pm" src="https://cloud.githubusercontent.com/assets/1019609/21297378/52e68448-c533-11e6-89ee-e3ec4d9db62f.png">

**Comment by torkelo** [comment]
@utkarshcmu yea, we wanted to tone it down a bit so it was not as attention grabbing. Do you think it's too hard to spot?

**Comment by dutchiechris** [comment]
I fetched again from Docker hub [ Grafana v4.1.0-pre1 (commit: 090db94) ] which looks like the most recent commit from this morning and still I see nothing after adding some help text:

![image](https://cloud.githubusercontent.com/assets/917241/21311871/bddb80bc-c5e9-11e6-9e3d-b9496d58cea4.png)

Ideas?  Should this feature be visible in the docker builds?

**Comment by torkelo** [comment]
@dutchiechris thanks for the heads up, looks like there is an issue with the css and minification which makes the panel help icon not show up (in the optimized build)

**Comment by utkarshcmu** [comment]
@torkelo Thats a good question whether the description should catch attention or not. Some users might want the "i" to be more brighter and some dimmer. Right now as a user its somewhat hard to spot it as its dimmer but I think either lets keep it this way or make it a bit(not much) brighter.

But this looks great, it will help a lot of users to know what information is exactly displayed in the panel. 👍

**Comment by torkelo** [comment]
@dutchiechris I fixed the css minification issue, next grafana/grafana:master docker image should have it working

**Comment by mattttt** [comment]
@utkarshcmu: Keep us posted as you use it, but as dashboards are all about information priority, we want to make sure we dont have everything competing for attention. We can play with some shades of grey, but once the user learns it, I think it's appropriately clear.

**Comment by kylebrandt** [comment]
FWIW (haven't tried the branch yet) but I was thinking about the ability to
have different icons, i.e. (exclamation, question mark) that dashboard
authors could choose.  However, that will make authors have to decide which
is correct when, and the idea was less appealing to me - but figured I
throw it out there.

**Comment by utkarshcmu** [comment]
Agreed with @mattttt 👍

**Comment by liujin1993** [comment]
@torkelo Which css file controls the panel help tooltip？I want to modify it to satisfy ourselves need.
